lib/tasks/search.rake

```diff
@@ -45,34 +45,86 @@ namespace :search do
 
-    nb_dataset_processed = 0
-    Dataset.published.find_in_batches(batch_size: 50) do |datasets|
-      puts " Batching #{datasets.length} datasets (done #{nb_dataset_processed} / #{nb_datasets_to_index})"
-      bulk_index(datasets)
-      nb_dataset_processed = nb_dataset_processed + 50
+    def bulk_index(datasets, new_index_name)
+      begin
+        Dataset.__elasticsearch__.client.bulk({
+          index: new_index_name,
+          type: ::Dataset.__elasticsearch__.document_type,
+          body: prepare_records(datasets)
+        })
+      rescue => e
+        puts "There was an error uploading datasets:"
+        puts e
+      end
     end
-  end
 
-  def bulk_index(datasets)
-    begin
-      Dataset.__elasticsearch__.client.bulk({
-                                              index: ::Dataset.__elasticsearch__.index_name,
-                                              type: ::Dataset.__elasticsearch__.document_type,
-                                              body: prepare_records(datasets)
-                                            })
-    rescue => e
-      puts "There was an error uploading datasets:"
-      puts e
+    def prepare_records(datasets)
+      datasets.map do |dataset|
+        {index: {_id: dataset.id, data: dataset.as_indexed_json}}
+      end
     end
-  end
 
-  def prepare_records(datasets)
-    datasets.map do |dataset|
-      {index: {_id: dataset.id, data: dataset.as_indexed_json}}
+
+    def index_datasets
+      @client.indices.create index: @new_index_name, body: {mappings: @dataset_mappings}
+      nb_dataset_processed = 0
+      Dataset.published.find_in_batches(batch_size: 50) do |datasets|
+        puts " Batching #{datasets.length} datasets (done #{nb_dataset_processed} / #{@nb_datasets_to_index})"
+        bulk_index(datasets, @new_index_name)
+        nb_dataset_processed += 50
+      end
     end
-  end
 
+    def update_alias
+      begin
+      @client.indices.update_aliases body: {
+        actions: [
+          {
+            remove: {
+              index: Dataset.index_name, alias: @alias
+            }          
+          }
+        ]
+      }
+      rescue=>e
+        puts 'No alias to remove'
+      end
+
+      @client.indices.update_aliases body: {
+        actions: [
+          {
+            add: {
+              index: @new_index_name, alias: @alias
+            }
+          }
+        ]
+      }
+      puts "Updated alias"
+    end
+
+    def delete_old_aliases
+      aliases = @client.indices.get_aliases.keys
+      aliases.each do |alias_name|
+        begin
+          if Time.parse(alias_name.gsub(/`datasets-development_`/, '')) < 1.weeks.ago
+            @client.indices.delete index: alias_name
+            puts "Deleted alias #{alias_name}"
+          end
+        rescue=>e
+          puts 'Failed to delete alias'
+          puts e.message
+        end
+      end
+      puts 'Import done'
+    end
+
+    index_datasets
+    update_alias
+    delete_old_aliases
```

Comment:
Currently it deletes any aliases older than a week but since we do not reindex that often perhaps I should instead keep the most recent 2 / 3 and delete any others?

Reply:
Will look at testing also